### ISSUE TITLE
`Naming/FileName` check reports `Rakefile.rb`

### ISSUE THREAD

**Issue by artur-intech** [rubocop/rubocop#12815]
My rake file is named `Rakefile.rb` and I get the following report:

```lang-none
rubocop Rakefile.rb
Inspecting 1 file
C

Offenses:

Rakefile.rb:1:1: C: Naming/FileName: The name of this source file (Rakefile.rb) should use snake_case.
```

Is there any reason why this is considered an issue? `Rakefile.rb` is the [default filename](https://github.com/ruby/rake/blob/v13.2.0/lib/rake/application.rb#L41) among the others.

Why those filenames are not listed in the [default config](https://github.com/rubocop/rubocop/blob/v1.62.1/config/default.yml#L2765)?

**Comment by bbatsov** [comment]
> Is there any reason why this is considered an issue? Rakefile.rb is the [default filename](https://github.com/ruby/rake/blob/v13.2.0/lib/rake/application.rb#L41) among the others.

Not really a problem, just we hadn't accounted for. I guess we'll have to update the default config, as I was aware only of `Rakefile` and `rakefile.rb`.

**Comment by bbatsov** [comment]
I see we also forgot to rename `Exclude` to `AllowedNames`/`AllowedFileNames` or something along those lines.

**Comment by koic** [comment]
Fixed in #12816.
